src/Microsoft.Identity.Client/Core/IdToken.cs

```diff
@@ -76,6 +77,9 @@ internal class IdToken
         [DataMember(Name = IdTokenClaim.HomeObjectId, IsRequired = false)]
         public string HomeObjectId { get; set; }
 
+        [DataMember(Name = IdTokenClaim.Upn, IsRequired = false)]
```

Comment:
@MarkZuber : to be considered in the cache schema

Reply:
@trwalke : pls. help making sure this one is followed up upon in our discussions around the schema. @MarkZuber?

---
In reply to: [255316867](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r255316867) [](ancestors = 255316867)

Reply:
@trwalke ?

---
In reply to: [255758876](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r255758876) [](ancestors = 255758876,255316867)

Reply:
@henrik-me Ill be sure to follow up on this

Reply:
Yes, Upn should be there and it's there in the C++ cache as well.  And this is the correct key for it.  So we're good here.